dipy/denoise/denspeed.pyx

```diff
@@ -117,8 +111,8 @@ def _nlmeans_3d(double[:, :, ::1] arr, double[:, :, ::1] mask,
 
                     out[i, j, k] = process_block(arr, i, j, k, B, P, sigma)
 
-    if have_openmp and num_threads is not None:
-        openmp.omp_set_num_threads(all_cores)
+    if num_threads is not None:
+        restore_default_num_threads()
```

Comment:
Yes, I think that would have been better. Resetting the number of threads to the default seems to me as unexpected as not restoring it at all. Anyway, that issue is unrelated to this PR =).